### PR TITLE
amam/setup release

### PR DESCRIPTION
**Pull request by mustofa-binary**
- add dockerfile
- add config nginx
- add circle ci config yml

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,236 @@
+version: 2.1
+orbs:
+    k8s: circleci/kubernetes@0.7.0
```
Comment (gokula-krishna-dev) [inline .circleci/config.yml]: We follow indentation with 2 space. Here it is 4

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+                      kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/<< parameters.k8s_svc_name >> << parameters.k8s_svc_name >>=${DOCKHUB_ORGANISATION}/derivcrypto-com:${CIRCLE_SHA1}
+
+jobs:
+    test:
+        docker:
+            - image: circleci/node:lts-stretch
+              environment:
+                  - GATSBY_CPU_COUNT=2
+                  - GATSBY_MAP_API_KEY=${GATSBY_MAP_API_KEY}
+                  - GATSBY_ENV=staging
```
Comment (gokula-krishna-dev) [inline .circleci/config.yml]: I thought we don't need these parameters anymore

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+                default: 'latest-staging'
+        steps:
+            - setup_remote_docker:
+                  docker_layer_caching: true
+            - run:
+                  name: Building docker image
+                  command: |
+                      docker build -t ${DOCKHUB_ORGANISATION}/derivcrypto-com:${CIRCLE_TAG} -t ${DOCKHUB_ORGANISATION}/derivcrypto-com:<< parameters.docker_image_tag >> .
+
+    docker_build_staging:
```
Comment (gokula-krishna-dev) [inline .circleci/config.yml]: This is repeated

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                default: 'staging-derivcrypto-com'
+        steps:
+            - k8s/install-kubectl
+            - run:
+                  name: Deploying to k8s cluster for service << parameters.k8s_svc_name >>
+                  command: |
+                      echo $CA_CRT | base64 --decode > ca.crt
+                      kubectl --server=${KUBE_SERVER} --certificate-authority=ca.crt --token=$SERVICEACCOUNT_TOKEN set image deployment/<< parameters.k8s_svc_name >> << parameters.k8s_svc_name >>=${DOCKHUB_ORGANISATION}/derivcrypto-com:${CIRCLE_TAG}
+
+    k8s_deploy_staging:
```
Comment (gokula-krishna-dev) [inline .circleci/config.yml]: This is repeated as well.

File: .circleci/config.yml (hunk 3 above)
Comment (gokula-krishna-dev) [reply .circleci/config.yml]: Call `docker_build` on jobs instead `docker_build_staging` to build for staging

**Review by gokula-krishna-dev — CHANGES_REQUESTED** [review]

**Review by gokula-krishna-dev — APPROVED** [review]

**Review by suthesank — APPROVED** [review]

**Pull request merged**